File: algorithms/342.py
# https://leetcode.com/problems/power-of-four


class Solution(object):
    def isPowerOfFour(self, num):
        """
        :type num: int
        :rtype: bool
        """
        while num >= 4 and num % 4 == 0:
            num /= 4
        return num == 1

File: algorithms/342.swift
// https://leetcode.com/problems/power-of-four

class Solution {
    func isPowerOfFour(num: Int) -> Bool {
        var n = num

        while (n >= 4 && n % 4 == 0) {
            n /= 4
        }
        return (n==1)
    }
}
